pkg/tdexdconnect/tdexdconnect_test.go

```diff
@@ -34,31 +34,34 @@ Wwq8JtrendvmccXB
 
 func TestEncodeDecode(t *testing.T) {
 	tests := []struct {
+		scheme      string
 		addr        string
 		cert        []byte
 		mac         []byte
 		expectedURL string
 	}{
 		{
+			scheme:      "https",
 			addr:        "localhost:9000",
 			cert:        []byte(certificate),
 			mac:         macaroon,
-			expectedURL: "tdexdconnect://localhost:9000?cert=MIICqDCCAk6gAwIBAgIRAM4trYu7swKO7E9OI5q350YwCgYIKoZIzj0EAwIwQjENMAsGA1UEChMEdGRleDExMC8GA1UEAxMoTUJQZGlQaXJhbGJlcnRvLmhvbWVuZXQudGVsZWNvbWl0YWxpYS5pdDAeFw0yMTEwMDYxNTMzMDRaFw0yMjEwMDcxNTMzMDRaMEIxDTALBgNVBAoTBHRkZXgxMTAvBgNVBAMTKE1CUGRpUGlyYWxiZXJ0by5ob21lbmV0LnRlbGVjb21pdGFsaWEuaXQwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAR1WPEU-MKQVHNAdK0bG89FGsU6Sw6Izqe1taOfS0hBQfP-W2YBccLwooIKmPdjtZAoV4kZcguksW6gAdzgtQrmo4IBIzCCAR8wDgYDVR0PAQH_BAQDAgKkMA8GA1UdEwEB_wQFMAMBAf8wHQYDVR0OBBYEFI5lIKaTMgXPe0v8k9oRakHlPHtoMIHcBgNVHREEgdQwgdGCKE1CUGRpUGlyYWxiZXJ0by5ob21lbmV0LnRlbGVjb21pdGFsaWEuaXSCCWxvY2FsaG9zdIIEdW5peIIKdW5peHBhY2tldIcEfwAAAYcQAAAAAAAAAAAAAAAAAAAAAYcQ_oAAAAAAAAAAAAAAAAAAAYcQ_oAAAAAAAAAQCEC_uSHHLIcEwKgB14cQ_oAAAAAAAACcVyn__lDMuocQ_oAAAAAAAADp7kZyxh-R8IcQ_oAAAAAAAABBBISbVTjXoYcQ_oAAAAAAAACu3kj__gARIjAKBggqhkjOPQQDAgNIADBFAiEAv-StshVAd-iSAz_2oGC0e076aiVvWHgKSehisPugrngCIGQ3tjiqzC1-oxNNMFvr7OD4CAkbWwq8JtrendvmccXB&macaroon=AgEFdGRleGQChQEDChCqCBfWunS_Tu_-IsVdx6XxEgEwGhUKBm1hcmtldBIEcmVhZBIFd3JpdGUaFwoIb3BlcmF0b3ISBHJlYWQSBXdyaXRlGg4KBXByaWNlEgV3cml0ZRoVCgZ3YWxsZXQSBHJlYWQSBXdyaXRlGhYKB3dlYmhvb2sSBHJlYWQSBXdyaXRlAAAGINX1smi9KiBJiORAbg2aY5wNJ0dk45Uz4Iy3gIyGHgUC",
+			expectedURL: "https://localhost:9000?cert=MIICqDCCAk6gAwIBAgIRAM4trYu7swKO7E9OI5q350YwCgYIKoZIzj0EAwIwQjENMAsGA1UEChMEdGRleDExMC8GA1UEAxMoTUJQZGlQaXJhbGJlcnRvLmhvbWVuZXQudGVsZWNvbWl0YWxpYS5pdDAeFw0yMTEwMDYxNTMzMDRaFw0yMjEwMDcxNTMzMDRaMEIxDTALBgNVBAoTBHRkZXgxMTAvBgNVBAMTKE1CUGRpUGlyYWxiZXJ0by5ob21lbmV0LnRlbGVjb21pdGFsaWEuaXQwWTATBgcqhkjOPQIBBggqhkjOPQMBBwNCAAR1WPEU-MKQVHNAdK0bG89FGsU6Sw6Izqe1taOfS0hBQfP-W2YBccLwooIKmPdjtZAoV4kZcguksW6gAdzgtQrmo4IBIzCCAR8wDgYDVR0PAQH_BAQDAgKkMA8GA1UdEwEB_wQFMAMBAf8wHQYDVR0OBBYEFI5lIKaTMgXPe0v8k9oRakHlPHtoMIHcBgNVHREEgdQwgdGCKE1CUGRpUGlyYWxiZXJ0by5ob21lbmV0LnRlbGVjb21pdGFsaWEuaXSCCWxvY2FsaG9zdIIEdW5peIIKdW5peHBhY2tldIcEfwAAAYcQAAAAAAAAAAAAAAAAAAAAAYcQ_oAAAAAAAAAAAAAAAAAAAYcQ_oAAAAAAAAAQCEC_uSHHLIcEwKgB14cQ_oAAAAAAAACcVyn__lDMuocQ_oAAAAAAAADp7kZyxh-R8IcQ_oAAAAAAAABBBISbVTjXoYcQ_oAAAAAAAACu3kj__gARIjAKBggqhkjOPQQDAgNIADBFAiEAv-StshVAd-iSAz_2oGC0e076aiVvWHgKSehisPugrngCIGQ3tjiqzC1-oxNNMFvr7OD4CAkbWwq8JtrendvmccXB&macaroon=AgEFdGRleGQChQEDChCqCBfWunS_Tu_-IsVdx6XxEgEwGhUKBm1hcmtldBIEcmVhZBIFd3JpdGUaFwoIb3BlcmF0b3ISBHJlYWQSBXdyaXRlGg4KBXByaWNlEgV3cml0ZRoVCgZ3YWxsZXQSBHJlYWQSBXdyaXRlGhYKB3dlYmhvb2sSBHJlYWQSBXdyaXRlAAAGINX1smi9KiBJiORAbg2aY5wNJ0dk45Uz4Iy3gIyGHgUC",
```

Comment:
I thought we wanted to add a new param to the URL like `proto=http|https`, instead we are replacing the initial `tdexdconnect://` with the scheme protocol right away. Is this intended? @tiero

Reply:
Yes this wrong indeed. We need a new querystring param `proto`

`tdexdconnect` must reamain as scheme

Reply:
to be clear this shoud be a complete valid URLs

- `tdexdconnect://<host>?proto=https&cert=<cert>&macaroon=<macaroon>`
- `tdexdconnect://<host>?proto=https&macaroon=<macaroon>`
- `tdexdconnect://<host>?proto=http&macaroon=<macaroon>`
- `tdexdconnect://<host>?proto=http`

Reply:
url.Sheme (pkg url) should to be used for protocol(ws, http, https)
@tiero why we use tdexdconnect ?

Reply:
and maybe also 

- `tdexdconnect://<host>?proto=https`
- `tdexdconnect://<host>?proto=http&cert=<cert>`

but not sure how valuable it is.

Reply:
> @tiero why we use tdexdconnect ?

This is our scheme we created, to make it clear is tdexdconnect URL. ALways has been

Reply:
example: https://github.com/LN-Zap/lndconnect/blob/master/lnd_connect_uri.md

This makes sure it's used by "TDEXD clients" and not normal http clients such as browsers, curls etc...

Reply:
you can use any scheme you want with pkg url

https://github.com/LN-Zap/lndconnect/blob/master/lndconnect.go#L71